src/openrct2-ui/windows/EditorInventionsList.cpp

```diff
@@ -218,17 +219,18 @@ static void research_scenery_groups_setup()
         if (entry_type != OBJECT_TYPE_SCENERY_GROUP)
             continue;
 
-        rct_research_item* research = gResearchItems;
-        for (; research->rawValue != RESEARCHED_ITEMS_END; research++)
-        {
-            if ((research->rawValue & 0xFFFFFF) != entryIndex)
-                continue;
+        auto SetAlwaysResearchedFlag = [entryIndex](rct_research_item& researchItem) {
+            if ((researchItem.rawValue & 0xFFFFFF) == entryIndex)
+            {
+                researchItem.flags |= RESEARCH_ENTRY_FLAG_SCENERY_SET_ALWAYS_RESEARCHED;
+                _editorInventionsListDraggedItem = &researchItem;
+                move_research_item(gResearchedResearchItems.data());
+                _editorInventionsListDraggedItem = nullptr;
+            }
+        };
 
-            research->flags |= RESEARCH_ENTRY_FLAG_SCENERY_SET_ALWAYS_RESEARCHED;
-            _editorInventionsListDraggedItem = research;
-            move_research_item(gResearchItems);
-            _editorInventionsListDraggedItem = nullptr;
-        }
+        std::for_each(std::begin(gResearchedResearchItems), std::end(gResearchedResearchItems), SetAlwaysResearchedFlag);
+        std::for_each(std::begin(gUnResearchedResearchItems), std::end(gUnResearchedResearchItems), SetAlwaysResearchedFlag);
```

Comment:
Can't we just use standard c++ for each instead of the stl for each.

Reply:
Agreed, a ranged for-loop would be much better here.